perf/terraform/region/main.tf

```diff
@@ -111,33 +111,42 @@ resource "aws_security_group" "restricted_inbound" {
   })
 }
 
+# If one wanted to make the key pair ephemeral, these would be the steps:
+# 1. Move this resource to `perf/terraform/ephemeral/main.tf`
+# 2. Add `make ssh-keygen` run step to `.github/workflows/perf.yml`
+# 3. Add `aws_key_pair` cleanup to `perf/terraform/common/files/scale_down.py`
+# 4. Allow `aws_iam_user.perf` from `perf/terraform/common/main.tf` to create/delete/update key pairs
+# 5. Allow `aws_iam_role.scale_down` from `perf/terraform/common/scale_down.tf` to delete key pairs
```

Comment:
> As far as I can tell, it introduces the need for a long lived SSH keypair through the AWS launch template.

We either have to have long lived SSH key pair or allow long lived AWS creds to create new AWS key pairs. With long lived SSH key pair, there's one fewer thing for us to clean up. I included the instructions here on how we could transition to ephemeral SSH keys.

Reply:
Thanks for writing this out.

> With long lived SSH key pair, there's one fewer thing for us to clean up.

AWS does not charge for SSH keys, right? Given that the clean-up step is just for times where we make mistakes, do we really need to clean them up in the lambda?

Without long-lived SSH keys we have one piece of state less to worry about.

All that said, I am fine either way.

Reply:
Yeah, it doesn't cost anything. I moved the SSH key generation and added TODOs for cleaning up the key pairs.